packages/stylist-core/src/style.rs

```diff
@@ -210,47 +206,10 @@ impl Style {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::ast::sample_scopes;
 
     #[test]
     fn test_simple() {
-        Style::new("background-color: black;").expect("Failed to create Style.");
-    }
-
-    #[test]
-    fn test_complex() {
-        let style = Style::new(
-            r#"
-                background-color: black;
-
-                .with-class {
-                    color: red;
-                }
-
-                @media screen and (max-width: 600px) {
-                    color: yellow;
-                }
-            "#,
-        )
-        .expect("Failed to create Style.");
-
-        assert_eq!(
-            style.get_style_str(),
-            format!(
-                r#".{style_name} {{
-background-color: black;
-}}
-.{style_name} .with-class {{
-color: red;
-}}
-
-@media screen and (max-width: 600px) {{
-.{style_name} {{
-color: yellow;
-}}
-}}
-"#,
-                style_name = style.get_class_name()
-            )
-        )
```

Comment:
TODO: add these tests back in `stylist` instead of `stylist-core`.

Reply:
I am not sure if Style also needs to be moved to `stylist_core`.

Does the procedural macro also need to use Style at compile time?

If my understanding is correct, it's the generated code inside `quote!` that will use Style to construct from AST.

Reply:
Good point, maybe `Style` and the registry should live in `stylist` instead of `stylist-core`, that would also solve an issue I had with documentation not showing the `impl From<Styles> for Classes` - which does show for `pub type Style = stylist_core::Style` but not if one uses `pub use stylist_core::Style`. Additionally could get rid of `StyleExt`.

In that case `style-core` would contain only the ast representation, the macro generates a `Sheet`, but not a `Style` and the user calls the correct `Style::new`/`Style::create`.

Reply:
I think you can generate code that references to the main crate (Style) as long as you are not using them at the compile time.

As the macro and core will be reexport to stylist, and generated code should also only reference them via the re-exported path in the main crate so they don't have to list stylist-core as a dependency.

Example: https://github.com/yewstack/yew/blob/master/packages/yew-macro/src/classes/mod.rs#L30

Please correct me if I am wrong.